ta/pkcs11/include/pkcs11_ta.h

```diff
@@ -565,8 +565,87 @@ enum pkcs11_ta_cmd {
 	 * This command relates to the PKCS#11 API function C_GenerateRandom().
 	 */
 	PKCS11_CMD_GENERATE_RANDOM = 42,
+
+	/*
+	 * PKCS11_CMD_DIGEST_INIT - Initialize a digest computation processing
+	 *
+	 * [in]  memref[0] = [
+	 *              32bit session handle,
+	 *              (struct pkcs11_attribute_head)mechanism +
+	 *                                            mechanism params,
+	 *	 ]
+	 * [out] memref[0] = 32bit return code, enum pkcs11_rc
+	 *
+	 * These commands relate to the PKCS#11 API function C_DigestInit().
+	 */
+	PKCS11_CMD_DIGEST_INIT = 44,
+
+	/*
+	 * PKCS11_CMD_DIGEST_KEY - Update digest with a key
+	 *
+	 * [in]  memref[0] = [
+	 *              32bit session handle,
+	 *              32bit key handle,
+	 *	 ]
+	 * [out] memref[0] = 32bit return code, enum pkcs11_rc
+	 *
+	 * These commands relate to the PKCS#11 API function C_DigestKey().
+	 */
+	PKCS11_CMD_DIGEST_KEY = 45,
+
+	/*
+	 * PKCS11_CMD_DIGEST_UPDATE - Update digest with data
+	 *
+	 * [in]  memref[0] = [
+	 *              32bit session handle,
+	 *              32bit command flags,
```

Comment:
Does digest update really need command flags?
there are no output buffer to query size from.

Reply:
Yes -- bad argument processing needs to be triggered somehow so that:
```
 A call to C_DigestUpdate which results in an error terminates the current digest operation.
```
(this bad arguments stuff was a bit of surprise)